include/tvm/arith/analyzer.h

```diff
@@ -107,6 +107,7 @@ class ConstIntBound : public ObjectRef {
  */
 class ConstIntBoundAnalyzer {
  public:
+  using BoundMapType = std::unordered_map<ObjectPtr<const PrimExprNode>, ConstIntBound, ObjectHash>;
```

Comment:
Let us just do std::unordered_map<PrimExpr, ConstIntBound, ObjectHash, ObjectEqual>;

Reply:
So, no const?  Then `GetObjectPtr` will need a `const_cast`, e.g.
```
auto op = GetElementPtr(const_cast<PrimExprNode*>(expr.as<PrimExprNode>()));
```

Do you also want to remove the first commit?

Reply:
Wait, maybe I misunderstood.  `PrimExpr` instead of `PrimExprNode`?

Reply:
yes, PrimExpr itself exposes the ptr as const pointer

Reply:
Using `PrimExpr` worked fine.  I tried `ObjectRef` before, but it failed (I guess I missed the ObjectHash/Equal for refs), and I went the way of `ObjectPtr`.  This really makes all the remaining  changes unnecessary (although I think the default return type of `GetObjectPtr` may be useful).

What do you suggest I do with the rest of the changes?  Should I remove the `ObjectPtr<const>` completely?

Reply:
I think it can go both ways, given that the most reference types already have const properties, perhaps we could remove `ObjectPtr<const>`, perhaps we can remove it for simplicity